db/crud.go

```diff
@@ -1165,6 +1165,26 @@ func (db *DatabaseCollectionWithUser) PutExistingRevWithBody(ctx context.Context
 
 }
 
+func (db *DatabaseCollectionWithUser) SyncFnDryrun(ctx context.Context, body Body, docID string) (*uint32, string, base.Set, channels.AccessMap, channels.AccessMap, error) {
+	doc := &Document{
+		ID:    docID,
+		_body: body,
+	}
+	// Make sure the user has permission to modify the document before confirming doc existence
+	mutableBody, metaMap, newRevID, err := db.prepareSyncFn(doc, doc)
+	if err != nil {
+		base.InfofCtx(ctx, base.KeyCRUD, "Failed to prepare to run sync function: %v", err)
```

Comment:
Can we change the log key here to a new diagnostic endpoint log key?  It will help us differentiate between diagnostic and non-diagnostic logging (particularly in cases like this where the log message is just cut-and-paste from non-diagnostic code).